src/h5web/vis-packs/core/shared/VisCanvas.tsx

```diff
@@ -3,23 +3,25 @@ import { Canvas } from '@react-three/fiber';
 import { useMeasure } from 'react-use';
 import styles from './VisCanvas.module.css';
 import type { AxisConfig } from '../models';
-import { computeVisSize } from '../utils';
+import { computeVisSize, computeSameRatioDomain } from '../utils';
 import AxisSystem from './AxisSystem';
 import AxisSystemContext from './AxisSystemContext';
+import type { Layout } from '../heatmap/models';
 
 const AXIS_OFFSETS = { vertical: 72, horizontal: 40, fallback: 16 };
 
 interface Props {
   abscissaConfig: AxisConfig;
   ordinateConfig: AxisConfig;
+  layout?: Layout;
   aspectRatio?: number;
   canvasTitle?: string;
   children: ReactNode;
 }
 
 function VisCanvas(props: Props) {
-  const { abscissaConfig, ordinateConfig, aspectRatio, canvasTitle, children } =
-    props;
+  const { aspectRatio, canvasTitle, children, layout = 'fill' } = props;
+  let { abscissaConfig, ordinateConfig } = props;
```

Comment:
I choose the most direct way for now for discussion by mutating the config domains if the `layout` is `loose`.

This ought to be changed once we agree on the roles of each component.